mcf-browser/src/TimelineExplorer.tsx

```diff
@@ -0,0 +1,364 @@
+            const locationMapping: any = {};
+            locationOptions.map(
+                (obj) => locationMapping[obj.value] = obj.label,
+            );
+            const page = 0;
+            const statVarGroups = Object.values(
+                this.groupByVariableMeasured(locations),
+            ) as Series[][];
+            return {
+              locations,
+              locationOptions,
+              selectKey,
+              locationMapping,
+              statVarGroups,
+              page,
+            };
+          });
+        },
+    );
+  }
+
+  /** Set the location options
+    * @param {TimelineExplorerPropType} prevProps the previous props
+   */
+  componentDidUpdate(prevProps: TimelineExplorerPropType) {
+    if (prevProps.data !== this.props.data) {
+      this.componentDidMount();
+    }
+  }
+
+  /** Returns the series objects which match the current locations filter
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Series[]} series that match the filter
+   */
+  filterLocations(locations: string[]) {
+    const cleanedLocations = new Set(
+        locations.map((location) =>
+         (location === 'undefined') ? undefined: location,
+        ),
+    );
+
+    const filteredData = this.props.data.filter(
+        (series) => cleanedLocations.has(series.observationAbout),
+    );
+
+    return filteredData;
+  }
+
+  /** Processes the data passed in by props and returns the
+    * data grouped by variableMeasured
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Object} an object mapping from variableMeasured to an array
+    * of Series with that variableMeasured value
+    */
+  groupByVariableMeasured(locations: string[]) {
+    const output: any = {};
+    // Filter by location
+    const data = this.filterLocations(locations);
+
+    for (const series of data) {
+      const varMeasured = series.variableMeasured ?
+         series.variableMeasured :
+         '';
+      if (!output[varMeasured]) {
+        output[varMeasured] = [];
+      }
+      output[varMeasured] = output[varMeasured].concat([series]);
+    }
+
+    return output;
+  }
+
+
+  /** Returns the JSX to render a group of related series
+    * @param {Series[]} seriesList a list of series objects with
+    *                              the same varMeasured
+    * @param {Object} locationMapping a mapping from location dcid to name
+    * @return {Object} a details element plotting all of the series
+   */
+  renderSeriesGroup(seriesList: Series[], locationMapping: Object) {
+    const varMeasured = seriesList[0].variableMeasured;
+    const groups = groupLocations(seriesList);
+    const groupNames = Object.keys(groups);
+
+    return (
+      <details className="stat-var-section" key={varMeasured}>
+        <summary>{varMeasured}</summary>
+        {
+          groupNames.map(
+              (groupName) =>
+                renderTimeGraph(
```

Comment:
It's not currently but I put it there since it doesn't explicitly rely on any code in the Timeline Explorer (e.g. doesn't use instance variables specific to TimelineExplorer), and could potentially be reused in the future for other explorers that might have TimeGraphs.

Reply:
I think we want to keep files as cohesive as possible & since renderTimeGraph and its helper functions aren't being shared, lets keep them in TimelineExplorer & they can be extracted as needed

Reply:
Sounds good!